crates/next-dev/src/lib.rs

```diff
@@ -199,51 +211,70 @@ impl NextDevServerBuilder {
             log_level: self.log_level,
         };
         let entry_requests = Arc::new(self.entry_requests);
-        let console_ui = Arc::new(ConsoleUi::new(log_options));
-        let console_ui_to_dev_server = console_ui.clone();
         let server_addr = Arc::new(server.addr);
         let tasks = turbo_tasks.clone();
+        let issue_reporter_once = Arc::new(Lazy::new(|| {
+            self.issue_reporter
+                .unwrap_or_else(|| {
+                    // Initialize a ConsoleUi reporter if no custom reporter was provided
+                    Box::new(|log_options| ConsoleUi::new(log_options).cell().into())
+                })
+                .get_issue_reporter(log_options)
+        }));
+        let issue_reporter_arc = Arc::new(move || **issue_reporter_once);
+        let get_issue_reporter = issue_reporter_arc.clone();
         let source = move || {
             source(
                 root_dir.clone(),
                 project_dir.clone(),
                 entry_requests.clone().into(),
                 eager_compile,
                 turbo_tasks.clone().into(),
-                console_ui.clone().into(),
+                get_issue_reporter(),
                 browserslist_query.clone(),
                 server_addr.clone().into(),
             )
         };
 
-        Ok(server.serve(tasks, source, console_ui_to_dev_server))
+        Ok(server.serve(tasks, source, issue_reporter_arc.clone()))
     }
 }
 
-async fn handle_issues<T: Into<RawVc>>(source: T, console_ui: ConsoleUiVc) -> Result<()> {
-    let state = console_ui
-        .group_and_display_issues(TransientValue::new(source.into()))
-        .await?;
+async fn handle_issues<T: Into<RawVc> + CollectiblesSource + Copy>(
+    source: T,
+    issue_reporter: IssueReporterVc,
+) -> Result<()> {
+    let issues = IssueVc::peek_issues_with_path(source).await?;
+    issue_reporter.report_issues(issues, TransientValue::new(source.into()));
+
+    let mut has_fatal = false;
+    for issue in issues.strongly_consistent().await?.iter() {
```

Comment:
The `strongly_consistent` needs to come before the call to `report_issues`, that way we get the same issues for both the reporting and the fatal iteration. We'll need to update `TaskInput` to receive a `ReadRef`, @alexkirsz might be able to help with that.